examples/gps_denied.py

```diff
@@ -254,121 +259,109 @@ def calculatePixelLocation(camera, map_keys, map_descs):
+    sleep(1)
+    environ["OPENCV_FFMPEG_CAPTURE_OPTIONS"] = "protocol_whitelist;file,rtp,udp"
+    cam = cv2.VideoCapture("./sololink.sdp")
+    return cam
 
 
 def take_picture(camera):
+    '''Takes a picture using the cimulated camera or the connected camera'''
     if CV_SIMULATION:
         return cv_simulation()
     else:
         try:
-            global img_counter
+            global IMG_COUNTER # pylint: disable=global-statement
             count = camera.get(cv2.CAP_PROP_FRAME_COUNT)
             camera.set(cv2.CAP_PROP_POS_FRAMES, count - 1)
             camera.grab()
             _, img = camera.retrieve()
-            cv2.imwrite("solopics/" + str(img_counter) + ".png", img)
+            cv2.imwrite("solopics/" + str(IMG_COUNTER) + ".png", img)
             crop_img = img[78:630, 270:1071]
-            img_counter += 1
+            IMG_COUNTER += 1
             return crop_img
-        except KeyboardError:
+        except KeyboardInterrupt:
             raise
-        except:
+        except TypeError:
             # Try taking the picture again
             sleep(1)
             take_picture(camera)
 
 
-def euclidean_distance(x1, y1, x2, y2):
-    return sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
+def euclidean_distance(x_initial, y_initial, x_final, y_final):
+    '''Calculates distance between two points'''
+    return sqrt((x_initial - x_final) ** 2 + (y_initial - y_final) ** 2)
 
 
-def change_yaw(vehicle, thetaDeg):
-    set_attitude(vehicle, yaw_angle=thetaDeg + vehicle.heading, duration=3)
+def move(vehicle, theta_deg, duration):
+    '''moves the vehicle in the direction thetat for duration seconds'''
+    theta_deg = theta_deg / 180.0 * pi
+    print("moving", theta_deg, sin(theta_deg), cos(theta_deg))
+    set_velocity(vehicle, x_velocity=cos(theta_deg) * MAX_VELOCITY,
+                 y_velocity=sin(theta_deg) * MAX_VELOCITY, duration=duration)
 
 
-def move_forward(vehicle, duration):
-    set_attitude(
-        vehicle, pitch_angle=-16, yaw_rate=0, use_yaw_rate=True, duration=duration
-    )
-
-
-def move(vehicle, thetaDeg, duration):
-    thetaDeg = thetaDeg / 180.0 * pi
-    print("moving", thetaDeg, sin(thetaDeg), cos(thetaDeg))
-    set_velocity(vehicle, vx = cos(thetaDeg) * MAX_VELOCITY, vy = sin(thetaDeg) * MAX_VELOCITY, duration = duration)
-
-
-#Sets the velocity of the vehicle RELATIVE TO HEADING
-def set_velocity(vehicle, vx = 0, vy = 0, duration = 0):
-    global X_VELOCITY
-    global Y_VELOCITY
-    xpid = PID(P, I, D, setpoint=vx * 133)
-    xpid.sample_time = 0.1
-    ypid = PID(P, I, D, setpoint=vy * 133)
-    ypid.sample_time = 0.1
-    for x in range(0, duration * 10):
+def set_velocity(vehicle, x_velocity=0, y_velocity=0, duration=0):
+    '''Sets the velocity of the vehicle RELATIVE TO HEADING'''
+    x_pid = PID(P, I, D, setpoint=x_velocity * 133)
+    x_pid.sample_time = 0.1
+    y_pid = PID(P, I, D, setpoint=y_velocity * 133)
+    y_pid.sample_time = 0.1
+    for _ in range(0, duration * 10):
```

Comment:
I think you forgot to add sleep(0.1) inside this loop.

Reply:
The sleep is included in the set_attitude call via the duration, that only returns after the duration.

Reply:
You right